jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java

```diff
@@ -930,18 +940,43 @@ public void reportFailures(FailureContext context) {
             }
         }
 
-        private void performStorySteps(RunContext context, List<PerformableSteps> storySteps, Stage stage)
+        private void performStorySteps(RunContext context, List<PerformableSteps> storySteps, Stage stage,
+                                       Optional<UUIDExceptionWrapper> causeToSkip)
                 throws InterruptedException {
             context.reporter().beforeStorySteps(stage);
             for (PerformableSteps steps : storySteps) {
-                steps.perform(context);
+                steps.perform(context, causeToSkip);
             }
             context.reporter().afterStorySteps(stage);
         }
 
-        private void performScenarios(RunContext context) throws InterruptedException {
+        private void performScenarios(RunContext context, Optional<UUIDExceptionWrapper> causeToSkip)
+                throws InterruptedException {
             for (PerformableScenario scenario : scenarios) {
-                scenario.perform(context);
+                scenario.perform(context, causeToSkip.isPresent() ? causeToSkip : causeToSkip(scenario));
+                putScenarioId(scenario);
+            }
+        }
+
+        private Optional<UUIDExceptionWrapper> causeToSkip(PerformableScenario scenario) {
+            Meta meta = scenario.getScenario().getMeta();
+            UUIDExceptionWrapper wrappedCause = null;
+            if (meta.hasProperty(DEPENDS_ON_ID)) {
+                String dependsOnId = meta.getProperty(DEPENDS_ON_ID);
+                Boolean performDependentScenario = successfulScenariosId.get(dependsOnId);
+                if (performDependentScenario == null || !performDependentScenario) {
+                    Throwable cause = performDependentScenario == null ? new DependsOnNotFoundException(dependsOnId)
+                            : new DependsOnFailedException(dependsOnId);
+                    wrappedCause = new UUIDExceptionWrapper(cause);
+                }
+            }
+            return Optional.ofNullable(wrappedCause);
```

Comment:
```java
Meta meta = scenario.getScenario().getMeta();
if (meta.hasProperty(DEPENDS_ON_ID)) {
    String dependsOnId = meta.getProperty(DEPENDS_ON_ID);
    Boolean performDependentScenario = successfulScenariosId.get(dependsOnId);
    if (performDependentScenario == null) {
        return Optional.of(new UUIDExceptionWrapper(new DependsOnNotFoundException(dependsOnId)));
    }
    if (!performDependentScenario) {
        return Optional.of(new UUIDExceptionWrapper(new DependsOnFailedException(dependsOnId)));
    }
}
return Optional.empty();
```
`Optional.of(new UUIDExceptionWrapper(...))` - can be moved to a separate method